cpu/esp32/periph/pm.c

```diff
@@ -92,8 +92,7 @@ void pm_reboot(void)
     }
 
     /* suspend and flush UARTs */
-    for (int i = 0; i < 3; ++i) {
-        REG_SET_BIT(UART_FLOW_CONF_REG(i), UART_FORCE_XOFF);
+    for (int i = 0; i < SOC_UART_NUM; ++i) {
```

Comment:
Not 
```suggestion
    for (int i = 0; i < UART_NUMOF; ++i) {
```
?

Reply:
Theoretically, one could use UART0 as `UART_DEV(0)` and UART2 as `UART_DEV(1)`. Since `uart_tx_wait_idle(i)` addresses the physical UART peripherals, this would wait only until UART0 and UART1. So it is better here, to iterate over all physical UART peripherals.

Reply:
But isn’t this a problem if the board does not use / init all physical UARTs?

Reply:
The problem is the order of logical `UART_DEV(n)` devices can differ from the order of perheral UARTs. If a board would define `UART2_TXD` and `UART2_RXD` symbols but not `UART1_TXD` and `UART1_RXD` (`UART0_TXD` and `UART0_RXD` are defined by default), the peripheral driver uses
- peripheral UART0 as `UART_DEV(0)`
- peripheral UART2 as `UART_DEV(1)`

and initializes them. The `UART_NUMOF` would be 2 in that case.

The `pm_reboot` doesn't now this mapping an calls `uart_tx_wait_idle`  for peripheral UARTi. If `UART_NUMOF` woulf be used for the iteration, it would only wait for UART0 and UART1 becoming idle but not for UART2.

Sure, we could add a static assert or preprocessor error that a board can't defined a different order for logical `UART_DEV(n)` but this would be another PR.

Reply:
Isn't there a way to map logical -> physical UART?

But this is only a problem if calling `uart_tx_wait_idle()` on a non-initialized UART can cause an issue.
If it perfectly fine to call `uart_tx_wait_idle()` without the UART being initialized / used, then the code is OK - just add a comment stating this then.

Reply:
> Isn't there a way to map logical -> physical UART?

Sure, there would be, but this would be a bigger change.

> But this is only a problem if calling `uart_tx_wait_idle()` on a non-initialized UART can cause an issue.

It is no problem to call `uart_tx_wait_idle` for an uninitialized UART. In fact, in the former implementation `uart_tx_wait_idle` was always called for UART2, even though UART2 was never used. Even more, ESP32 SoCs that do not have UART2 worked with the former implementation.